tools/update-test-manifests

```diff
@@ -0,0 +1,96 @@
+
+
+def find_snapshot_date(distro, arch, schutzfile):
+    # Each distro can have multiple lists of repositories. We assume all share the same snapshot date, so we only check
+    # the first one.
+    repos = schutzfile[distro]["repos"][0][arch]
+    first_url = repos[0]["baseurl"]
+    *_, date = first_url.split("-")
+    return date
+
+
+def main():
+    if os.getuid() != 0:
+        print("This script requires root to build manifests", file=sys.stderr)
+        sys.exit(1)
+
+    vars_files = glob("test/data/manifests/*-vars.ipp.yaml")
+
+    schutzfile = read_schutzfile()
+
+    yaml = YAML(pure=True)
+    yaml.width = 9999
+    yaml.indent(sequence=4, offset=2)
+
+    for vf in vars_files:
+        vars_file = pathlib.Path(vf)
+        with vars_file.open(mode="r", encoding="utf-8") as vfp:
+            vars_data = yaml.load(vfp)
+
+        # NOTE: this relies on the filename to discover the distro, which isn't perfect because the file names can be
+        # arbitrary, but realistically we always name them based on the distro
+        distro_name, *_ = vars_file.parts[-1].split("-")
+        if distro_name == "centos":
+            distro_name += "-stream"
+        distro_ver = vars_data["mpp-vars"]["release"]
+        arch = vars_data["mpp-vars"]["arch"]
+        distro = f"{distro_name}-{distro_ver}"
+        vars_date = vars_data["mpp-vars"]["snapshot"]
+
+        if distro not in schutzfile:
+            print(f"Distro {distro} for ipp.vars file {vars_file} not found in Schutzfile.", file=sys.stderr)
+            print("Failed to update ipp.vars files.", file=sys.stderr)
+            sys.exit(1)
+
+        schutzfile_date = find_snapshot_date(distro, arch, schutzfile)
+        if vars_date == schutzfile_date:
+            print(f"{vars_file}:\n  Snapshot dates match: {vars_date}")
+            continue
+
+        print(f"{vars_file}:\n  Updating snapshots from {vars_date} to {schutzfile_date}")
+        vars_data["mpp-vars"]["snapshot"] = schutzfile_date
+
+        with vars_file.open(mode="w", encoding="utf-8") as vfp:
+            yaml.dump(vars_data, vfp)
+
+    # regenerate all test manifests (requires root)
+    sp.check_call(["make", "test-data"])
+
+    # find modified manifests
+    difflist = sp.check_output(["git", "diff", "--name-only", "test/data/stages"],
+                               encoding="utf-8").strip().splitlines()
+    modified_paths = set()
+    for diffpath in difflist:
+        modified_paths.add(os.path.dirname(diffpath))
+
+    # update stage test diffs (requires root)
+    with TemporaryDirectory() as tmpdir:
+        for stage_test in modified_paths:
+            newdiff = sp.check_output(["./tools/gen-stage-test-diff", "--libdir=.", f"--store={tmpdir}", stage_test],
+                                      encoding="utf-8")
+            diff_file = os.path.join(stage_test, "diff.json")
+            with open(diff_file, mode="w", encoding="utf-8") as fp:
+                fp.write(newdiff)
```

Comment:
In general I like `pathlib` more; and in this case you could `Path(diff_file).write_text(newdiff)` (or turn `diff_file` into a Path earlier).